### PR TITLE
added better RHEL support & mock tests

### PR DESCRIPTION
**Pull request by rvagg**

### CODE REVIEW

File: tests/mocktests.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+fs.stat = function(file, callback) {
+  process.nextTick(function() {
+    if (!currentData.file[file])
+      return callback(new Error())
+    callback(null, { isFile: function() { return true }})
+  })
+}
+
+fs.readFile = function(file, enc, callback) {
+  process.nextTick(function() {
```
Comment (wblankenship) [inline tests/mocktests.js]: What is `process.nextTick()` doing here? Is this to make the call truly async?

Comment (rvagg) [inline tests/mocktests.js]: always good practice to make functions that take callbacks actually be async so the user doesn't need to care about the order of call:

```js
fs.readFile('foo', function (err, data) {
  console.log('I read the file')
})
console.log('I am about to read the file')
```

```
I am about to read the file
I read the file
```
vs

```
I read the file
I am about to read the file
```

Comment (wblankenship) [inline tests/mocktests.js]: Ah, makes sense.

**Comment by wblankenship** [comment]
LGTM :+1:

**Comment by rvagg** [comment]
`npm version patch && npm publish`? (or `npm version minor` if you like)

**Pull request merged**